docs/libs/ml/quickstart.md

```diff
@@ -24,4 +24,198 @@ under the License.
+dataset with the FlinkML classification algorithms.
+
+{% highlight scala %}
+
+val cancerLV = cancer
+  .map(_.productIterator.toList)
+  .filter(!_.contains("?"))
+  .map{list =>
+    val numList = list.map(_.asInstanceOf[String].toDouble)
+    LabeledVector(numList(11), DenseVector(numList.take(10).toArray))
+    }
+
+{% endhighlight %}
+
+We can then use this data to train a learner.
+
+A common format for ML datasets is the LibSVM format and a number of datasets using that format can be
+found [in the LibSVM datasets website](http://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/). FlinkML provides utilities for loading
+datasets using the LibSVM format through the `readLibSVM` function available through the MLUtils object.
+You can also save datasets in the LibSVM format using the `writeLibSVM` function.
+Let's import the Adult (a9a) dataset. You can download the 
+[training set here](http://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/binary/a8a)
+and the [test set here](http://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/binary/a8a.t).
+
+We can simply import the dataset then using:
+
+{% highlight scala %}
+
+val adultTrain = MLUtils.readLibSVM("path/to/a8a")
+val adultTest = MLUtils.readLibSVM("path/to/a8a.t")
+
+{% endhighlight %}
+
+This gives us a `DataSet[LabeledVector]` that we will use in the following section to create a classifier.
+
+Due to an error in the test dataset we have to adjust the test data using the following code, to 
+ensure that the dimensionality of all test examples is 123, as with the training set:
+
+{% highlight scala %}
+
+val adjustedTest = adultTest.map{lv =>
+      val vec = lv.vector.asBreeze
+      val padded = vec.padTo(123, 0.0).toDenseVector
+      val fvec = padded.fromBreeze
+      LabeledVector(lv.label, fvec)
+    }
+
+{% endhighlight %}
+
+## Classification
+
+Once we have imported the dataset we can train a `Predictor` such as a linear SVM classifier.
+
+{% highlight scala %}
+
+val svm = SVM()
+  .setBlocks(env.getParallelism)
+  .setIterations(100)
+  .setRegularization(0.001)
+  .setStepsize(0.1)
+  .setSeed(42)
+
+svm.fit(adultTrain)
+
+{% endhighlight %}
+
+Let's now make predictions on the test set and see how well we do in terms of absolute error
+We will also create a function that thresholds the predictions to the {-1, 1} scale that the
+dataset uses.
+
+{% highlight scala %}
+
+def thresholdPredictions(predictions: DataSet[(Double, Double)])
+: DataSet[(Double, Double)] = {
+  predictions.map {
+    truthPrediction =>
+      val truth = truthPrediction._1
+      val prediction = truthPrediction._2
+      val thresholdedPrediction = if (prediction > 0.0) 1.0 else -1.0
+      (truth, thresholdedPrediction)
+  }
+}
+
+val predictionPairs = thresholdPredictions(svm.predict(adjustedTest))
+
+val absoluteErrorSum = predictionPairs.collect().map{
+  case (truth, prediction) => Math.abs(truth - prediction)}.sum
```

Comment:
Shouldn't we divide the error by two? Otherwise each error will add `2` to the absolute error sum.